### ISSUE TITLE
g10k fails on some yaml configs that are valid and work with r10k

### ISSUE THREAD

**Issue by gdubicki** [xorpaul/g10k#41]
g10k fails to parse the following config correctly:

```
---
:cachedir: '/var/cache/r10k'
:sources:
  :hieradata:
    remote: git@git.mydomain.com:puppet/hiera.git
    basedir: /etc/puppet/hieradata
  :envs:
    remote: git@git.mydomain.com:puppet/envs.git
    basedir: /etc/puppet/environments
```

...with the following output:

```
Synced /etc/puppetlabs/r10k/r10k.yaml with 0 git repositories and 0 Forge modules in 0.0s with git (0.0s sync, I/O 0.0s) and Forge (0.0s query+download, I/O 0.0s)
```

That config with removed some of `:`s, like this:

```
---
cachedir: '/var/cache/r10k'
sources:
  hieradata:
    remote: git@git.mydomain.com:puppet/hiera.git
    basedir: /etc/puppet/hieradata
  envs:
    remote: git@git.mydomain.com:puppet/envs.git
    basedir: /etc/puppet/environments
```

...works correctly in g10k.

Note that the first one is valid YAML, according to http://www.yamllint.com/ AND is correctly parsed by r10k.

Of course it was easy to resolve, but it was not so easy to come up with the reason g10k is not working, so perhaps it's worth to do something about it.

**Comment by maxadamo** [comment]
my two cents.
while the first one has a valid yaml syntax, I would say that it has a key name set to `:cachedir` (starting with colon), instead of `cachedir` in the second case. 
IMO, I don't see a reason to match a value starting with colon, where the colone (as of yaml syntax) is part of the name. 
I tried to translate the yaml to a ruby hash and this is what I've seen. If r10k worked the way around too, it was not correct.

**Comment by maxadamo** [comment]
to be clear, you can try to put as many colon you want, and the belong to the name (I use `pry`):
`
require 'yaml'
[4] pry(main)> hash = YAML.load(File.read("test.yaml"))
=> {:"ca:c:hedir"=>"/var/cache/r10k",
`

**Comment by xorpaul** [comment]
Yeah, I'm using the standard golang YAML module gopkg.in/yaml.v2. Looks like it doesn't like the `:foobar:` format. I'll try to debug tomorrow.

**Comment by maxadamo** [comment]
> Maybe I can just always substitute the first : in every config value. 

I think this is what they are doing. It's nice to have for fully compatibility. but it didn't make much sense (to me)

**Comment by gdubicki** [comment]
Shouldn't 0 git repos and 0 forge modules be treated as error? Then just print a friendly message to check you config for subtle (or not so subtle) yaml isses and note that some r10k config don't work without changes and we're done with it.

(Actually the second issue I reported today bugs me more as it prevents me to test g10k without changes in my codebase. So I would be greatful if you could look at it instead, @xorpaul :))

**Comment by xorpaul** [comment]
I could equate 0 git repos and 0 forge modules found as an error, but what if someone doesn't use modules and has all Puppet code directly in the environment Git repository? That wouldn't work then.

**Comment by xorpaul** [comment]
#43 should fix this, but I feel somewhat dirty now :/

https://github.com/xorpaul/g10k/pull/43/files

**Comment by xorpaul** [comment]
closed with #43
